kolena/workflow/visualization/_activation_map.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC
+from abc import abstractmethod
+
+import numpy as np
+
+from kolena.errors import InputValidationError
+
+
+class ColorMap(ABC):
+    @abstractmethod
+    def red(self, intensity: np.uint8) -> np.uint8:
+        ...
+
+    @abstractmethod
+    def green(self, intensity: np.uint8) -> np.uint8:
+        ...
+
+    @abstractmethod
+    def blue(self, intensity: np.uint8) -> np.uint8:
+        ...
+
+    def _interpolate(self, intensity: np.uint8, x0: float, y0: float, x1: float, y1: float) -> np.uint8:
+        return np.uint8(round((intensity - x0) * (y1 - y0) / (x1 - x0) + y0))
+
+
+class ColorMapJet(ColorMap):
+    """
+    The [MATLAB "Jet" color palette](http://blogs.mathworks.com/images/loren/73/colormapManip_14.png) is a standard
+    palette used for scientific and mathematical data.
+
+    It is defined as a linear ramp between the following colours: "#00007F", "blue", "#007FFF", "cyan", "#7FFF7F",
+    "yellow", "#FF7F00", "red", "#7F0000"
+    """
+
+    def _scale_to_colormap(self, intensity: np.uint8) -> np.uint8:
+        max_uint8 = np.iinfo(np.uint8).max
+        if intensity <= max_uint8 / 8:
+            return 0
+        elif intensity <= max_uint8 * 3 / 8:
+            return self._interpolate(intensity, max_uint8 / 8, 0.0, max_uint8 * 3 / 8, max_uint8)
+        elif intensity <= max_uint8 * 5 / 8:
+            return max_uint8
+        elif intensity <= max_uint8 * 7 / 8:
+            return self._interpolate(intensity, max_uint8 * 5 / 8, max_uint8, max_uint8 * 7 / 8, 0.0)
+        else:
+            return 0
+
+    def red(self, intensity: np.uint8) -> float:
+        return self._scale_to_colormap(intensity - np.iinfo(np.uint8).max / 4)
+
+    def green(self, intensity: np.uint8) -> float:
+        return self._scale_to_colormap(intensity)
+
+    def blue(self, intensity: np.uint8) -> float:
+        return self._scale_to_colormap(intensity + np.iinfo(np.uint8).max / 4)
+
+
+def colorize_activation_map(activation_map: np.ndarray, fade_low_activation: bool = True) -> np.ndarray:
```

Comment:
Not critical for now but it seems to me an improved interface would be to have `colormap` as a kwarg (w/ default) and bake the `fade_low_activation` into the colormap. That allows users to plug in their own colors, if desired, and keeps this function simple.

Reply:
Ah I like that. Thanks for the suggestion. Will make that change.